src/workloads/execution/lookup/LookupNLJ.yml

```diff
@@ -0,0 +1,345 @@
+    Key: NLJLookupCmdTemplate
+    Parameters:
+      l: *LocalDoc100k
+      f: *Foreign10K
+      lk: "int_key2"
+      k: "arr8"
+
+NLJLookupForeign10kArr8ScalarCmd: &NLJLookupForeign10kArr8ScalarCmd
+  LoadConfig:
+    Path: "../../phases/execution/LookupCommands.yml"
+    Key: NLJLookupCmdTemplate
+    Parameters:
+      l: *LocalDoc100k
+      f: *Foreign10K
+      lk: "arr8"
+      k: "int_key2"
+
+# Defines test operations for 'NLJLookup' actors.
+NLJTestOperations: &NLJTestOperations
+- OperationMetricsName: Foreign1k_Scalar_Scalar
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign1kScalarScalarCmd
+- OperationMetricsName: Foreign1k_Scalar_Object2
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign1kScalarObj2Cmd
+- OperationMetricsName: Foreign1k_Scalar_Object5
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign1kScalarObj5Cmd
+- OperationMetricsName: Foreign1k_Object5_Scalar
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign1kObj5ScalarCmd
+- OperationMetricsName: Foreign1k_Scalar_Array1
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign1kScalarArr1Cmd
+- OperationMetricsName: Foreign1k_Scalar_Array8
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign1kScalarArr8Cmd
+- OperationMetricsName: Foreign1k_Array8_Scalar
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign1kArr8ScalarCmd
+- OperationMetricsName: Foreign10k_Scalar_Scalar
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign10kScalarScalarCmd
+- OperationMetricsName: Foreign10k_Scalar_Object2
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign10kScalarObj2Cmd
+- OperationMetricsName: Foreign10k_Scalar_Object5
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign10kScalarObj5Cmd
+- OperationMetricsName: Foreign10k_Object5_Scalar
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign10kObj5ScalarCmd
+- OperationMetricsName: Foreign10k_Scalar_Array1
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign10kScalarArr1Cmd
+- OperationMetricsName: Foreign10k_Scalar_Array8
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign10kScalarArr8Cmd
+- OperationMetricsName: Foreign10k_Array8_Scalar
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign10kArr8ScalarCmd
+
+# Defines how many times each test operation is executed.
+TestRepeatCount: &TestRepeatCount 10
+
+Actors:
+# The 'Setup' actor drops the database and runs in the 0th phase.
+- Name: Setup
+  Type: RunCommand
+  Database: *db
+  Thread: 1
+  Phases:
+  - Repeat: 1
+    Threads: 1
+    Database: *db
+    Operations:
+    - OperationName: RunCommand
+      OperationCommand: {dropDatabase: 1}
+  - Phase: 1..3
+    Nop: true
+
+# The 'LoadForeign1K' / 'LoadForeign10K' / 'LoadLocalDoc100k' actors run in the 1st phase.
+- Name: LoadForeign1K
+  Type: CrudActor
+  Database: *db
+  Threads: 1
+  Phases:
+  - &Nop {Nop: true}
```

Comment:
Instead of running 'insertOne' 1k or 10k times, could this use 'insertMany' or BulkWrite ops? Something like so: 
https://github.com/mongodb/genny/blob/master/src/workloads/docs/CrudActor.yml#L37-L42 or https://github.com/mongodb/genny/blob/master/src/cast_core/include/cast_core/actors/CrudActor.hpp#L60-L67 ?

Reply:
I believe the insertMany and BulkWrite op only insert the listed documents once, not repeatedly one document many times. So I have to list all 10k documents in order to use one insertMany, or I can list 10 documents and repeat it 1k times, I don't think this is better than using insertOne.